source/extensions/filters/http/grpc_field_extraction/extractor_impl.cc

```diff
@@ -0,0 +1,63 @@
+#include "source/extensions/filters/http/grpc_field_extraction/extractor_impl.h"
+
+#include <algorithm>
+#include <iterator>
+#include <string>
+#include <vector>
+
+#include "source/common/common/logger.h"
+
+#include "absl/strings/str_format.h"
+#include "proto_field_extraction/field_value_extractor/field_value_extractor_factory.h"
+#include "proto_field_extraction/field_value_extractor/field_value_extractor_interface.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcFieldExtraction {
+namespace {
+
+using Protobuf::field_extraction::FieldValueExtractorFactory;
+using Protobuf::field_extraction::FieldValueExtractorInterface;
+
+} // namespace
+
+absl::Status ExtractorImpl::init() {
+  FieldValueExtractorFactory extractor_factory(type_finder_);
```

Comment:
nit: consider dependency injecting this for testability. That way you can mock `FieldValueExtractorFactory`

But maybe not needed as you can easily test with real `FieldValueExtractorFactory` dependency

Reply:
Yeah prefer to not mock if it is feasible test with the real implementation.